decode.go

```diff
@@ -112,21 +136,24 @@ func (dec *Decoder) SetBufferSize(n int) {
 	dec.buf = bytes.NewBuffer(make([]byte, 0, n))
 }
 
+// Get the file header
+func (dec *Decoder) Header() (*Header, error) {
```

Comment:
I believe this may be simplified to the following without any worries about race conditions:
```
func (dec *Decoder) Header() (*Header, error) {
    return dec.header, dec.readOSMHeader()
}
```
The `dec.header` pointer should always be passed to the calling function after `dec.readOSMHeader()` has been executed so `dec.header` will be set unless `dec.readOSMHeader()` produces an error. The down side of my suggestion is that the race condition, or lack thereof, is less explicit.